kinder/pkg/cluster/manager/actions/setup-external-ca.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package actions
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"k8s.io/kubeadm/kinder/pkg/cluster/status"
+)
+
+// SetupExternalCA setups certificates and kubeconfig files to be able to create a cluster without CA keys.
+func SetupExternalCA(c *status.Cluster, vLevel int) error {
+	fmt.Println("Setuping external CA for the cluster...")
+
+	// gets the IP of the load balancer
+	loadBalancerIP, _, err := c.ExternalLoadBalancer().IP()
+	if err != nil {
+		return errors.Wrapf(err, "failed to get IP for node: %s", c.ExternalLoadBalancer().Name())
+	}
+
+	// generate certs on the primary node
+	if err := c.BootstrapControlPlane().Command(
+		"/bin/sh", "-c",
+		fmt.Sprintf("kubeadm init phase certs all --control-plane-endpoint=%s --v=%d",
+			loadBalancerIP,
+			vLevel),
+	).RunWithEcho(); err != nil {
+		return errors.Wrapf(err, "could not generate certs on node: %s", c.BootstrapControlPlane().Name())
+	}
+
+	// copy the shared certs to the rest of the CP nodes
+	for _, n := range c.SecondaryControlPlanes() {
+		if err := copyCertificatesToNode(c, n); err != nil {
+			return errors.Wrapf(err, "could not copy certs to node %s", n.Name())
+		}
+	}
+
+	// generate kubeconfig files on the primary node
+	if err := c.BootstrapControlPlane().Command(
+		"/bin/sh", "-c",
+		fmt.Sprintf("kubeadm init phase kubeconfig all --control-plane-endpoint=%s --v=%d",
+			loadBalancerIP,
+			vLevel),
+	).RunWithEcho(); err != nil {
+		return errors.Wrapf(err, "could not generate kubeconfig files on node: %s", c.BootstrapControlPlane().Name())
+	}
+
+	// copy the shared kubeconfig files to the rest of the CP nodes
+	for _, n := range c.SecondaryControlPlanes() {
+		if err := copyKubeconfigFilesToNode(c, n); err != nil {
+			return errors.Wrapf(err, "could not copy kubeconfig files to node %s", n.Name())
+		}
+	}
+
+	// generate remaining certificates on the secondary CP nodes
+	for _, n := range c.SecondaryControlPlanes() {
+		if err := n.Command(
+			"/bin/sh", "-c",
+			fmt.Sprintf("kubeadm init phase certs all --control-plane-endpoint=%s --v=%d",
+				loadBalancerIP,
+				vLevel),
+		).RunWithEcho(); err != nil {
+			return errors.Wrapf(err, "could not generate remaining certificates on node: %s", n.Name())
+		}
+	}
```

Comment:
might be readability for this logic could be improved if we generate first things on the bootstrap node, then we executing a unique loop on secondary cp with copy certs, copy kubeconfig, generate missing certs